thebus.go

```diff
@@ -0,0 +1,160 @@
+	"strings"
+)
+
+type Arrivals struct {
+	StopTimes struct {
+		Timestamp string `json:"timestamp"`
+		Arrival   []struct {
+			Headsign  string `json:"headsign"`
+			Vehicle   string `json:"vehicle"`
+			StopTime  string `json:"stopTime"`
+			Estimated string `json:"estimated"`
+			Latitude  string `json:"latitude"`
+			Shape     string `json:"shape"`
+			ID        string `json:"id"`
+			Trip      string `json:"trip"`
+			Canceled  string `json:"canceled"`
+			Date      string `json:"date"`
+			Longitude string `json:"longitude"`
+			Route     string `json:"route"`
+			Direction string `json:"direction"`
+		} `json:"arrival"`
+		Stop string `json:"stop"`
+	} `json:"stopTimes"`
+}
+
+type Vehicle struct {
+	Vehicles struct {
+		Timestamp string `json:"timestamp"`
+		Vehicle   struct {
+			Driver         string `json:"driver"`
+			Longitude      string `json:"longitude"`
+			LastMessage    string `json:"last_message"`
+			Headsign       string `json:"headsign"`
+			RouteShortName string `json:"route_short_name"`
+			Number         string `json:"number"`
+			Trip           string `json:"trip"`
+			Latitude       string `json:"latitude"`
+			Adherence      string `json:"adherence"`
+		} `json:"vehicle"`
+	} `json:"vehicles"`
+}
+
+type Routes struct {
+	Routes struct {
+		RouteName string `json:"routeName"`
+		RouteID   string `json:"routeID"`
+		Route     []struct {
+			Headsign  string `json:"headsign"`
+			RouteNum  string `json:"routeNum"`
+			ShapeID   string `json:"shapeID"`
+			FirstStop string `json:"firstStop"`
+		} `json:"route"`
+	} `json:"routes"`
+}
+
+func fetchArrivals(s string) (*Arrivals, error) {
+	arrivalsPath := fmt.Sprintf("http://api.thebus.org/arrivals/?key=%v&stop=%v", apiKey, s)
+	resp, err := http.Get(arrivalsPath)
+	if err != nil {
+		log.Print(err)
+	}
+	defer resp.Body.Close()
+
+	respData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Print(err)
+	}
+
+	respString := string(respData)
+	xmlString := strings.NewReader(respString)
+
+	arrivals := &Arrivals{}
+
+	jsonResp, err := xj.Convert(xmlString)
+	if err != nil {
+		log.Print(err)
+	}
+
+	if err := json.Unmarshal(jsonResp.Bytes(), arrivals); err != nil {
+		log.Print(err)
+	}
+
+	return arrivals, nil
+}
+
+func fetchVehicle(s string) (*Vehicle, error) {
+	vehiclePath := fmt.Sprintf("http://api.thebus.org/vehicle/?key=%v&num=%v", apiKey, s)
+	resp, err := http.Get(vehiclePath)
+	if err != nil {
+		log.Print(err)
+	}
+	defer resp.Body.Close()
+
+	respData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Print(err)
+	}
+
+	respString := string(respData)
+	xmlString := strings.NewReader(respString)
+
+	vehicle := &Vehicle{}
+
+	jsonResp, err := xj.Convert(xmlString)
+	if err != nil {
+		log.Print(err)
+	}
+
+	if err := json.Unmarshal(jsonResp.Bytes(), vehicle); err != nil {
+		log.Print(err)
+	}
+
+	return vehicle, nil
+}
+
+func fetchRoutes(s string, t string) (*Routes, error) {
+	validRouteType := map[string]bool{"route": true, "headsign": true}
```

Comment:
Are there other/better ways to implement a `oneOf` value check for this argument?